samples/subscriptions/createPushSubscription.js

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019-2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on
+ * subscriptions with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+'use strict';
+
+// Creates a new push subscription.
+function main(
+  topicName = 'YOUR_TOPIC_NAME',
+  subscriptionName = 'YOUR_SUBSCRIPTION_NAME'
+) {
+  // [START pubsub_create_push_subscription]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const topicName = 'YOUR_TOPIC_NAME';
+  // const subscriptionName = 'YOUR_SUBSCRIPTION_NAME';
+
+  // Imports the Google Cloud client library
+  const {PubSub} = require('@google-cloud/pubsub');
+
+  // Creates a client; cache this for further use
+  const pubSubClient = new PubSub();
+
+  async function createPushSubscription() {
+    const options = {
+      pushConfig: {
+        // Set to an HTTPS endpoint of your choice. If necessary, register
+        // (authorize) the domain on which the server is hosted.
+        pushEndpoint: `https://${pubSubClient.projectId}.appspot.com/push`,
+      },
+    };
+
+    await pubSubClient
+      .topic(topicName)
+      .createSubscription(subscriptionName, options);
+    console.log(`Subscription ${subscriptionName} created.`);
+  }
+
+  createPushSubscription();
```

Comment:
Should we add error handling to all the samples? The old samples used top level async functions, where as with this format if an error were to occur it would result in an unhandled rejection (I think). It is probably ok, but I'm not sure if we care about that scenario or not

Reply:
The updated style guide isn't super clear on that point, but it does seem like something we should do. It should be super easy with the way everything is being routed through sampleMain now, so I could just throw that in.

Reply:
I spent a while thinking about this and pondering the samples style guide, and I think there isn't a way to fulfill both. The style guide is pretty explicit that samples should have an async function which is immediately called at the end of the sample, but given that, there isn't a simple way to let exceptions/rejections bubble up. So I think for now I might just pass on fixing this. With Node 12 it gives a warning about unhandled rejections, but it does still print a stack trace, so that probably covers it for now...

Reply:
I changed my mind on this. It's still not 100% in line with the style guide, but it makes sense as part of a sample snippet anyway. In the call to the async "meat" function, I'm just adding a .catch(console.error) to print any errors.